### PR TITLE
Fix wrong LB idletimeout for node outbound

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**: In cluster scope, when reading Control Plane outbound LB specs, IdleTimeoutInMinutes is read from node outbound LB spec instead from control plane outbound spec https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/4964920c7467073ff445e6d971e78d98bc60f177/azure/scope/cluster.go#L287, so this can cause a panic on the mentioned line in cluster scope when node outbound LB is nil.

Adds unit tests for the LBSpecs() func.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Fix idleTimeoutInMinutes for Control Plane Outbound LB to use the right LB spec
```

### CODE REVIEW

**Review by sonasingh46 — APPROVED** [review]
PR looks good to me. 
Re-triggered the failed test, timeout issue.

**Comment by nprokopic** [comment]
Looks great, thanks!

**Comment by CecileRobertMichon** [comment]
Could I get an `/lgtm` from one of you?

/assign @mboersma @jackfrancis

**Review by mboersma — COMMENTED** [review]
/lgtm

🐛  Thank you for the unit test!

**Comment by CecileRobertMichon** [comment]
/approve

self-approving since 1) this is a relatively small change (outside UTs) and fixes a user impacting bug 2) 3 people have reviewed and lgtm'ed including one project approver and one project reviewer

**Pull request merged**
